lib/hurley/connection.rb

```diff
@@ -76,13 +75,13 @@ def net_http_request(request)
 
     def perform_request(http, request, res)
       if :get == request.verb
-        # prefer `get` to `request` because the former handles gzip (ruby 1.9)
-        http_res = http.get(request.url.request_uri, request.header.to_hash) do |chunk|
-          res.receive_body(chunk)
+        http.request_get(request.url.request_uri, request.header.to_hash) do |http_res|
+          res.status_code = http_res.code.to_i
+          http_res.read_body {|chunk| res.receive_body(chunk)}
```

Comment:
Any idea if `#request_get` handles gzip too?

This is nit picky, but I prefer spaces around the block separators:  `{ |chunk| res.receive_body(chunk) }`